api/files/views.py

```diff
@@ -34,15 +34,23 @@ class FileMixin(object):
     serializer_class = FileSerializer
     file_lookup_url_kwarg = 'file_id'
 
-    def get_file(self, check_permissions=True):
+    def get_file_obj(self, only_404=True):
         try:
-            obj = utils.get_object_or_error(BaseFileNode, self.kwargs[self.file_lookup_url_kwarg], self.request)
-        except (NotFound, Gone):
```

Comment:
@fabmiz, I think this entire fix might be as simple as not excepting `Gone` here so that it's actually raised. I really don't think `get_object_or_error` or `get_serializer_class` should need to be changed for this fix. 

Can you reevaluate and see if you agree?

Reply:
@caseyrollins, the reason why we need these changes is `get_serializer_class`. It needs the node object to determine if it returns `QuickFilesDetailSerializer` or `FileDetailSerializer`. Now, if we raise an error, be it `404` or `410`, in any call from within `get_serializer_class`, it translates to `502` (seems to be an issue with bapi/drf), which was the original issue. In the case of `404`, it does not really matter since there is the object does not exist, so we can return whatever serializer or the default. However in the case of `410`, `get_serializer_class` can get the object and return the right serializer. This is my current understanding. 

However, we could not except `410` in `get_file` as you suggest, except both errors and return a serializer we choose, since it does really matter.